### ISSUE TITLE
Support HDF filters affecting reading

### ISSUE THREAD

**Issue by JohnLCaron** [JohnLCaron/cdm-kotlin#117]
Quantization not neded for readers?

https://github.com/Unidata/netcdf-c/blob/main/docs/quantize.md

**Comment by JohnLCaron** [comment]
https://support.hdfgroup.org/documentation/hdf5-docs/hdf5_topics/UsingCompressionInHDF5.html

HDF5 supports gzip (deflate), Szip, n-bit, scale-offset, and shuffling (with deflate) compression filters. 

See the [Datasets chapter](https://hdfgroup.github.io/hdf5/develop/_h5_d__u_g.html#sec_dataset) of the HDF5 User’s Guide and the [Filters in HDF5](https://hdfgroup.github.io/hdf5/develop/_f_i_l_t_e_r.html) for more information on Data Pipeline Filters.

**Comment by lesserwhirls** [comment]
HDF also has a filter registry with many more options:

https://github.com/HDFGroup/hdf5_plugins/blob/master/docs/RegisteredFilterPlugins.md#list-of-filter-plugins-registered-with-the-hdf-group

**Comment by JohnLCaron** [comment]
From the Readme:

"There will be no dependencies on native libraries in the core module, but other modules or
projects that use the core are free to use dependencies as needed. We will likely add runtime discovery to facilitate this, 
for example, to use HDF5 filters that link to native libraries."

**Comment by JohnLCaron** [comment]
PR #170 adds filters from jhdf, using a simple "register filter" call, with interface
```
interface H5filterIF {
    fun id() : Int
    fun name() : String
    fun apply(encodedData: ByteArray, clientValues: IntArray): ByteArray
}
```
We have to write some kind of kotlin wrapper for any native filter.  Not sure of the advantage of "runtime filter discovery" thats more than just explitly registering?

**Comment by JohnLCaron** [comment]
Good point. There has to be check whether the library is loaded, with an error message rather than an exception.

Also, things should work as long as the filter is registered, ie, the filter doesnt have to be in our FilterType enumeration. As long as the filter id matches the one in the data file.
